python-package/xgboost/core.py

```diff
@@ -1160,7 +1160,7 @@ def feature_names(self, feature_names: Optional[FeatureNames]) -> None:
                 raise ValueError(msg)
             # prohibit to use symbols may affect to parse. e.g. []<
             if not all(isinstance(f, str) and
-                       not any(x in f for x in set(('[', ']', '<')))
+                       not any(x in f for x in {'[', ']', '<'})
```

Comment:
```suggestion
                       not any(x in f for x in ['[', ']', '<'])
```

`pylint` rightly caught that since this is iterator, not doing a set comparison, an iterable type should be used here instead of a set.

>  /home/runner/work/xgboost/xgboost/python-package/xgboost/core.py:1163: convention (C0208, use-sequence-for-iteration, DMatrix.feature_names) Use a sequence type when iterating over values

build link: https://github.com/dmlc/xgboost/actions/runs/3635213908/jobs/6134104007

Reply:
hmm, the change seems to be causing failure in tests.

Reply:
Interesting, I'm not sure why. I can't see how iterating over a list of strings should be different (in a way that matters for this code) than iterating over a set of strings.

I just merged latest `master` into this branch, in case the test failures aren't related to this PR's changes. If they still fail, I'll try to reproduce this locally.

Reply:
Ok yes, I was able to reproduce the failing tests locally.

```shell
cd python-package
pip install .
cd ..
pytest tests/python/test_early_stopping.py
```

I'll try to figure out which of these changes is to blame, sorry about that!